smart-py/objkt_swap.py

```diff
@@ -17,135 +38,679 @@ def __init__(self, objkt, hdao, manager, metadata, curate):
+            self.data.swaps[params].objkt_id,
+            self.data.swaps[params].objkt_amount
+        )
+
         del self.data.swaps[params]
-        
+
     @sp.entry_point
     def collect(self, params):
-        sp.verify( (params.objkt_amount > 0) & (sp.sender != self.data.swaps[params.swap_id].issuer) )
-        
+        sp.verify(
+            (params.objkt_amount > 0) &
+            # (params.objkt_amount == 1) &
+            (sp.sender != self.data.swaps[params.swap_id].issuer)
+        )
+
         sp.if (self.data.swaps[params.swap_id].xtz_per_objkt != sp.tez(0)):
-        
-            self.objkt_amount = sp.fst(sp.ediv(sp.amount, self.data.swaps[params.swap_id].xtz_per_objkt).open_some())
-            
-            self.amount = self.objkt_amount * sp.fst(sp.ediv(self.data.swaps[params.swap_id].xtz_per_objkt, sp.mutez(1)).open_some())
-            
-            sp.verify((params.objkt_amount == self.objkt_amount) & (sp.amount == sp.utils.nat_to_mutez(self.amount)) & (sp.amount > sp.tez(0)))
+
+            self.objkt_amount = sp.fst(
+                sp.ediv(
+                    sp.amount,
+                    self.data.swaps[params.swap_id].xtz_per_objkt
+                ).open_some()
+            )
+
+            self.amount = self.objkt_amount * sp.fst(
+                sp.ediv(
+                    self.data.swaps[params.swap_id].xtz_per_objkt,
+                    sp.mutez(1)
+                ).open_some()
+            )
+
+            sp.verify(
+                (params.objkt_amount == self.objkt_amount) &
+                (sp.amount == sp.utils.nat_to_mutez(self.amount)) &
+                (sp.amount > sp.tez(0))
+            )
+
             # calculate fees and royalties
-            self.fee = sp.fst(sp.ediv(sp.utils.nat_to_mutez(self.amount), sp.utils.nat_to_mutez(1)).open_some()) * (self.data.royalties[self.data.swaps[params.swap_id].objkt_id].royalties + 25) / 1000
+            self.fee = sp.fst(
+                sp.ediv(
+                    sp.utils.nat_to_mutez(self.amount),
+                    sp.utils.nat_to_mutez(1)).open_some()
+                ) * (self.data.royalties[self.data.swaps[params.swap_id].objkt_id].royalties + 25) / 1000
+
             self.royalties = self.data.royalties[self.data.swaps[params.swap_id].objkt_id].royalties * self.fee / (self.data.royalties[self.data.swaps[params.swap_id].objkt_id].royalties + 25)
-            
+
             # send royalties to NFT creator
-            sp.send(self.data.royalties[self.data.swaps[params.swap_id].objkt_id].issuer, sp.utils.nat_to_mutez(self.royalties))
-            
+            sp.send(
+                self.data.royalties[self.data.swaps[params.swap_id].objkt_id].issuer,
+                sp.utils.nat_to_mutez(self.royalties)
+            )
+
             # send management fees
-            sp.send(self.data.manager, sp.utils.nat_to_mutez(abs(self.fee - self.royalties)))
-            
+            sp.send(
+                self.data.manager,
```

Comment:
I wonder if the manager and the fees administrator should be a different person... That will facilitate work delegation.